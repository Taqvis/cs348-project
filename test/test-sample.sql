-- Delete all values in every non-track table
DELETE FROM Playlist_Songs;
DELETE FROM Playlists;
DELETE FROM Users;

-- Feature 3.i: Create account
INSERT INTO Users VALUES ("testuser", "Test User", "testpass");

SELECT * FROM Users WHERE username = "testuser";

-- Feature 3.ii: Login
SELECT COUNT(*) FROM Users WHERE username="testuser" AND password="testpass";

-- Feature 2i: Create a playlist
INSERT INTO Playlists VALUES ("testuser", "Test Playlist");

SELECT * FROM Playlists WHERE username = "testuser" AND playlist_name = "Test Playlist";

-- Feature 2ii: Rename a playlist
UPDATE Playlists SET playlist_name="New Test Playlist" WHERE playlist_name="Test Playlist"  AND username="testuser";

SELECT * FROM Playlists WHERE username = "testuser";

-- Feature 2iii: Add song to playlist
INSERT INTO Playlist_Songs VALUES ("testuser", "New Test Playlist", "02shCNmb6IvgB5jLqKjtkK");
INSERT INTO Playlist_Songs VALUES ("testuser", "New Test Playlist", "4GroumsjzAcgCwJzTGY7px");

-- Feature 2iv: Delete a song from the playlist
DELETE FROM Playlist_Songs WHERE username="testuser" AND playlist_name="New Test Playlist" AND track_id="4GroumsjzAcgCwJzTGY7px";

SELECT track_id, track_name FROM Playlist_Songs NATURAL JOIN Tracks WHERE username = "testuser" AND playlist_name = "New Test Playlist";

-- Feature 1i: Search song by Track name
SELECT * FROM Tracks WHERE track_name LIKE "%Nevada%";

SELECT * FROM Track_Artists WHERE EXISTS (SELECT track_id FROM Tracks WHERE Tracks.track_id = Track_Artists.track_id AND Tracks.track_name LIKE "%Nevada%");

-- Feature 1ii: Searh playlist by name
SELECT * FROM Playlists WHERE playlist_name LIKE "%New Test Playlist%";
