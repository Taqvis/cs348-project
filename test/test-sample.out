mysql> source C:\Users\Jesse\Desktop\loo\CS348\test-sample.sql
Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

Query OK, 1 row affected (0.01 sec)

Query OK, 1 row affected (0.00 sec)

+----------+--------------+----------+
| username | display_name | password |
+----------+--------------+----------+
| testuser | Test User    | testpass |
+----------+--------------+----------+
1 row in set (0.00 sec)

+----------+
| COUNT(*) |
+----------+
|        1 |
+----------+
1 row in set (0.00 sec)

+------------------------+------------+------------+------------+-------------+----------+-------------------+-------------------+-----------+--------------------+------+-------------------+-------------------+-------------------+-------------------+-------------------+---------+----------------+-------------+---------------+------------+-------------+
| track_id               | album_name | track_name | popularity | duration_ms | explicit | danceability      | energy            | music_key | loudness           | mode | speechiness       | acousticness      | instrumentalness  | liveness          | valence           | tempo   | time_signature | track_genre | accousticness | instrument | time_signal |
+------------------------+------------+------------+------------+-------------+----------+-------------------+-------------------+-----------+--------------------+------+-------------------+-------------------+-------------------+-------------------+-------------------+---------+----------------+-------------+---------------+------------+-------------+
| 02shCNmb6IvgB5jLqKjtkK | Nevada     | Nevada     |         71 |      208561 |        0 | 0.605000019073486 | 0.782999992370606 |         3 | -4.270999908447266 |    1 | 0.032400000840425 | 0.299000000000000 | 0.000175000000000 | 0.133000001311302 | 0.104999996721745 | 124.015 |              4 | edm         |          NULL |       NULL |        NULL |
+------------------------+------------+------------+------------+-------------+----------+-------------------+-------------------+-----------+--------------------+------+-------------------+-------------------+-------------------+-------------------+-------------------+---------+----------------+-------------+---------------+------------+-------------+
1 row in set (0.00 sec)

+------------------------+------------------+
| track_id               | artist           |
+------------------------+------------------+
| 02shCNmb6IvgB5jLqKjtkK | Cozi Zuehlsdorff |
| 02shCNmb6IvgB5jLqKjtkK | Vicetone         |
+------------------------+------------------+
2 rows in set (0.00 sec)

Query OK, 1 row affected (0.00 sec)

+----------+---------------+
| username | playlist_name |
+----------+---------------+
| testuser | Test Playlist |
+----------+---------------+
1 row in set (0.00 sec)

Query OK, 1 row affected (0.01 sec)
Rows matched: 1  Changed: 1  Warnings: 0

+----------+-------------------+
| username | playlist_name     |
+----------+-------------------+
| testuser | New Test Playlist |
+----------+-------------------+
1 row in set (0.00 sec)

Query OK, 1 row affected (0.00 sec)

Query OK, 1 row affected (0.00 sec)

Query OK, 1 row affected (0.00 sec)

+------------------------+------------+
| track_id               | track_name |
+------------------------+------------+
| 02shCNmb6IvgB5jLqKjtkK | Nevada     |
+------------------------+------------+
1 row in set (0.00 sec)

mysql> notee
